tensorflow/contrib/eager/python/examples/generative_examples/dcgan.ipynb

```diff
@@ -327,142 +431,134 @@
-        "* Last step is to plot the predictions and **voila!**"
+        "**Define training parameters**"
       ]
     },
     {
-      "cell_type": "code",
-      "execution_count": 0,
       "metadata": {
-        "colab": {},
         "colab_type": "code",
-        "id": "NS2GWywBbAWo"
+        "id": "NS2GWywBbAWo",
+        "colab": {}
       },
-      "outputs": [],
+      "cell_type": "code",
       "source": [
-        "EPOCHS = 150\n",
+        "EPOCHS = 50\n",
         "noise_dim = 100\n",
         "num_examples_to_generate = 16\n",
         "\n",
-        "# keeping the random vector constant for generation (prediction) so\n",
-        "# it will be easier to see the improvement of the gan.\n",
+        "# We'll re-use this random vector used to seed the generator so\n",
+        "# it will be easier to see the improvement over time.\n",
         "random_vector_for_generation = tf.random_normal([num_examples_to_generate,\n",
         "                                                 noise_dim])"
-      ]
+      ],
+      "execution_count": 0,
+      "outputs": []
     },
     {
-      "cell_type": "code",
-      "execution_count": 0,
       "metadata": {
-        "colab": {},
-        "colab_type": "code",
-        "id": "RmdVsmvhPxyy"
+        "colab_type": "text",
+        "id": "jylSonrqSWfi"
       },
-      "outputs": [],
+      "cell_type": "markdown",
       "source": [
-        "def generate_and_save_images(model, epoch, test_input):\n",
-        "  # make sure the training parameter is set to False because we\n",
-        "  # don't want to train the batchnorm layer when doing inference.\n",
-        "  predictions = model(test_input, training=False)\n",
+        "**Define training method**\n",
         "\n",
-        "  fig = plt.figure(figsize=(4,4))\n",
-        "  \n",
-        "  for i in range(predictions.shape[0]):\n",
-        "      plt.subplot(4, 4, i+1)\n",
-        "      plt.imshow(predictions[i, :, :, 0] * 127.5 + 127.5, cmap='gray')\n",
-        "      plt.axis('off')\n",
-        "        \n",
-        "  plt.savefig('image_at_epoch_{:04d}.png'.format(epoch))\n",
-        "  plt.show()"
+        "We start by iterating over the dataset. The generator is given a random vector as an input which is processed to  output an image looking like a handwritten digit. The discriminator is then shown the real MNIST images as well as the generated images.\n",
+        "\n",
+        "Next, we calculate the generator and the discriminator loss. Then, we calculate the gradients of loss with respect to both the generator and the discriminator variables."
       ]
     },
     {
-      "cell_type": "code",
-      "execution_count": 0,
       "metadata": {
-        "colab": {},
         "colab_type": "code",
-        "id": "2M7LmLtGEMQJ"
+        "id": "2M7LmLtGEMQJ",
+        "colab": {}
       },
-      "outputs": [],
+      "cell_type": "code",
```

Comment:
Why pass `noise_dim` as an argument to `train()` given that it is defined as almost a constant above?
The module level `noise_dim` constant and the function argument can be confusing. How about just having a `NOISE_DIM` constant and using that instead of creating a constant and then providing that as an argument of the same name to `train()`?

Reply:
Good catch. Fixed.